coulomb-physical-constants/src/main/scala/coulomb/physicalconstants/package.scala

```diff
@@ -0,0 +1,191 @@
+package coulomb
+
+import coulomb.define.DerivedUnit
+import coulomb.si._
+import coulomb.mks._
+import spire.math.Rational
+import spire.math.ConvertableTo
+
+package object physicalconstants {
+  import infra._
+  import units._
+
+  // The wikipedia link below is used as an authoritative source for constants
+  // https://en.wikipedia.org/wiki/List_of_physical_constants
+  //
+  // The name for each individual constant is taken directly from the link above
+  // to avoid ambiguites
+
+  def speedOfLightInVacuum[V](implicit pcq: PhysicalConstantQuantity[V, SpeedOfLightInVacuum]): Quantity[V, pcq.QU] = pcq.q
+
+  def planckConstant[V](implicit pcq: PhysicalConstantQuantity[V, PlanckConstant]): Quantity[V, pcq.QU] = pcq.q
+
+  def reducedPlanckConstant[V](implicit pcq: PhysicalConstantQuantity[V, ReducedPlanckConstant]): Quantity[V, pcq.QU] = pcq.q
+
+  def newtonianConstantOfGravitation[V](implicit pcq: PhysicalConstantQuantity[V, NewtonianConstantOfGravitation]): Quantity[V, pcq.QU] = pcq.q
+
+  def vacuumElectricPermittivity[V](implicit pcq: PhysicalConstantQuantity[V, VacuumElectricPermittivity]): Quantity[V, pcq.QU] = pcq.q
+
+  def vacuumMagneticPermeability[V](implicit pcq: PhysicalConstantQuantity[V, VacuumMagneticPermeability]): Quantity[V, pcq.QU] = pcq.q
+
+  def characteristicImpedanceOfVacuum[V](implicit pcq: PhysicalConstantQuantity[V, CharacteristicImpedanceOfVacuum]): Quantity[V, pcq.QU] = pcq.q
+
+  def elementaryCharge[V](implicit pcq: PhysicalConstantQuantity[V, ElementaryCharge]): Quantity[V, pcq.QU] = pcq.q
+
+  def hyperfineTransitionFrequencyOfCs133[V](implicit pcq: PhysicalConstantQuantity[V, HyperfineTransitionFrequencyOfCs133]): Quantity[V, pcq.QU] = pcq.q
+
+  def avogadroConstant[V](implicit pcq: PhysicalConstantQuantity[V, AvogadroConstant]): Quantity[V, pcq.QU] = pcq.q
+
+  def boltzmannConstant[V](implicit pcq: PhysicalConstantQuantity[V, BoltzmannConstant]): Quantity[V, pcq.QU] = pcq.q
+
+  def conductanceQuantum[V](implicit pcq: PhysicalConstantQuantity[V, ConductanceQuantum]): Quantity[V, pcq.QU] = pcq.q
+
+  def josephsonConstant[V](implicit pcq: PhysicalConstantQuantity[V, JosephsonConstant]): Quantity[V, pcq.QU] = pcq.q
+
+  def vonKlitzingConstant[V](implicit pcq: PhysicalConstantQuantity[V, VonKlitzingConstant]): Quantity[V, pcq.QU] = pcq.q
+
+  def magneticFluxQuantum[V](implicit pcq: PhysicalConstantQuantity[V, MagneticFluxQuantum]): Quantity[V, pcq.QU] = pcq.q
+
+}
+
+package physicalconstants {
+
+  // actual unit defs in a subpackage
+  object units {
+    // define physical constants as derived units
+    trait SpeedOfLightInVacuum
+    implicit def defineSpeedOfLightInVacuum =
```

Comment:
is there a reason you are creating the `DerivedUnit` instances using `def` (dynamic) instead of `val` or `object` (static)

Reply:
I guess it helps to use `denom` and `num`

Reply:
I could change the to use val, I guess it will be slightly more efficient
I confronted how to create big numbers and didn't want to use doubles to avoid precision loss hence I turned to represent numbers using `Rational` operations

Reply:
`Rational` is how I prefer to do it.  Defer any precision loss to the "end" when the final numeric type is populated.